pkg/query-service/app/http_handler.go

```diff
@@ -1869,6 +1871,50 @@ func (aH *APIHandler) setTTL(w http.ResponseWriter, r *http.Request) {
 
 }
 
+func (aH *APIHandler) setCustomRetentionTTL(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	claims, errv2 := authtypes.ClaimsFromContext(ctx)
+	if errv2 != nil {
+		RespondError(w, &model.ApiError{Err: errors.New("failed to get org id from context"), Typ: model.ErrorInternal}, nil)
+		return
+	}
+
+	var params model.CustomRetentionTTLParams
+	if err := json.NewDecoder(r.Body).Decode(&params); err != nil {
+		RespondError(w, &model.ApiError{Typ: model.ErrorBadData, Err: err}, nil)
+		return
+	}
+
+	// Context is not used here as TTL is long duration DB operation
+	result, apiErr := aH.reader.SetCustomRetentionTTL(context.Background(), claims.OrgID, &params)
+	if apiErr != nil {
+		if apiErr.Typ == model.ErrorConflict {
+			aH.HandleError(w, apiErr.Err, http.StatusConflict)
+		} else {
+			aH.HandleError(w, apiErr.Err, http.StatusInternalServerError)
+		}
+		return
+	}
+
+	aH.WriteJSON(w, r, result)
+}
+
+func (aH *APIHandler) getCustomRetentionTTL(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
```

Comment:
since we are using the same table, if someone has used the new API to set TTL, then won't it cause an issue if the existing API is called.

i.e shouldn't we add something more from the DB side to restrict i.e the old one shouldn't read the values added by the new one as here the logic of hot and cold ttl is completely different from the exisitng one ?